Server/src/server/gameCenter/models/game/Player.java

```diff
@@ -91,6 +92,35 @@ public void addCardToDeck(ServerCard card) throws LogicException {
         return drawnCards;
     }
 
+    public ServerCard[] getCardsFromDeckNoDuplicate(int cardsToDraw, ServerCard card) {
+        ServerCard[] drawnCards = new ServerCard[cardsToDraw];
+        Random RNGenerator = new Random();
+        int infiniteRedrawFailSafe = 10;
+
+		for(int i = 0; i < cardsToDraw; i++) {
+            if (!deck.getCards().isEmpty()) {
+                int index = RNGenerator.nextInt(deck.getCards().size());
+                ServerCard drawnCard = deck.getCards().get(index);
+                if( drawnCard.IDSameAs(card.getCardId()) && infiniteRedrawFailSafe > 0){
+                    System.out.println("Same card found, redrawing...");
+                    infiniteRedrawFailSafe--;
+                    i--;
```

Comment:
why decrement i here?    If it has the same id we don't want to draw the card (unless we have no other choice, of course).

Reply:
the idea here was if for the ith iteration we draw the same card, we decrement i and skip this loop so that we draw another card for the same ith iteration (i.e. i = 3, we draw the same card, decrement i down to 2, we skip this loop increasing i back to 3, and draw again). but after taking a break from the code i realized this could be achieved with a while loop.